src/server.h

```diff
@@ -34,6 +34,9 @@
 # include <QtConcurrent>
 # include <QFutureSynchronizer>
 #endif
+#ifndef _WIN32
+# include <sched.h>
```

Comment:
For what purpose do we need this new header?

Reply:
For the `#ifndef SCHED_RR` check.  `global.h` pulls in `unistd.h`, so I might be better protecting this include with `#ifdef POSIX` or something.

Reply:
Why not using #ifdef POSIX at the place where you want to avoid setting the "TimeCriticalPriority"?

Reply:
Because I was planning deep, dark magic if SCHED_RR were defined to use pthread_setpriority.  I've not got that working, though.

Reply:
Ok, it sound like there is some more work todo. Maybe it is a good idea that I create a branch from master called "multithreading_tests" where I merge your code to so that we have some kind of "playground". The reason why I am a bit careful is because changing thread priorities could have unwanted side effects. I would like to change thread priorities only if we can say for sure that the server really runs better/more stable with the new code.

Reply:
Definitely - with the new QtConcurrent model, I'm wondering if it's worth re-doing the QThread approach entirely.  QtConcurrent::run() seems fairly similar to moving a task to a thread and calling QThread::start() on that thread -- but much clearer.  I'll continue reading about it to see if I can get a clear idea on the QtConcurrent view on priority.

Reply:
Ok, I have just created a new branch for multithreading_improvements and will merge your branch in that one.

Reply:
> Because I was planning deep, dark magic if SCHED_RR were defined to use pthread_setpriority. I've not got that working, though.

Are you still progressing on that? Or is it on hold? I ask because I want to get rid of the # include<sched.h> for now. If you plan to continue your implementation, you may add it later on. So again, would it make sense to replace SCHED_RR by POSIX?